pkg/route/api/validation/validation.go

```diff
@@ -111,8 +115,94 @@ func ValidateRouteStatusUpdate(route *routeapi.Route, older *routeapi.Route) fie
 	return allErrs
 }
 
+type blockVerifierFunc func(block *pem.Block) error
+
+func publicKeyBlockVerifier(block *pem.Block) error {
```

Comment:
modifying in-place is a little unexpected... I'd expect a new block returned. there are other bits (block headers, etc) that this isn't clearing